app/Actions/Photo/Strategies/AddStrategyParameters.php

```diff
@@ -2,23 +2,25 @@
 
 namespace App\Actions\Photo\Strategies;
 
-use App\Actions\Photo\Extensions\SourceFileInfo;
+use App\Metadata\Extractor;
 use App\Models\Album;
 use App\Models\Configs;
 
 class AddStrategyParameters
 {
 	public ImportMode $importMode;
 
-	// Information about intended parent album
+	/** @var Album|null the intended parent album */
```

Comment:
Why the `/** @var Album|null` ?  Isn't the type in php enough ?

Reply:
If the question is meant as "should it be theoretically sufficient to add the type hint to the variable in an optimal world?" Yes, I agree. :grinning: Giving the type as a type hint in the code and as an type annotation in the comment introduces unnecessary redundancy. 

Unfortunately, this is how PHPdoc expects it. I did not want to loose the original comment and attach it to the variable. So I changed to `//` into a `/**`. But in order to make the comment actually appear in the documentation, I had to add the `@var` keyword.

If one looks at https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/var.html#var how the syntax is defined, all three components of the `@var` tag - type, name, description - seem to be optionally. But if you use the `@var` keyword and want to use the description you must add the type and the name, too. Otherwise PHPdoc thinks that the first word of the description is the type and the second one the name.

Examples:

Does not work:

```php
/** intended parent album */
public ?Album parent
```

Makes PHpdoc falsely assume that "intended" is the actual type, "parent" the name and "album" the description. :roll_eyes:

```php
/** @var intended parent album */
public ?Album parent
```

Actually works, but occupies too many lines for my personal taste. This tiny, little variable gets more space in the source code than it deserves :grinning:

```php
/**
 * Intended parent album
 * 
  *@var 
  */
public ?Album parent
```

This works, too, is a succinct one-liner but at the costs of redundancy

```php
/** @var Album|null Intended parent album */
public ?Album parent
```

Hence, I decided to got with the last solution, especially as my IDE keeps the type hint and the type annotation in sync for me. :yum: